apps/extension/src/ui/domains/Sign/SignRequestContext/EthereumSignMessageRequestContext.ts

```diff
@@ -44,7 +43,8 @@ const useEthSignMessageRequestProvider = ({ id }: KnownSigningRequestIdOnly<"eth
       // for now only check signTypedData's verifying contract's address
       const typedMessage = isTypedData ? JSON.parse(request.request) : undefined
       const verifyingContract = typedMessage?.domain?.verifyingContract as string | undefined
-      if (verifyingContract && !isEthereumAddress(verifyingContract)) return false
+      if (verifyingContract && verifyingContract.toLowerCase().startsWith("javascript:"))
```

Comment:
AFAICT all the other instances of the verifyingContract derive from this hook - so, it should be safe to validate here only (or else, upstream of here eg. in the tabs handler) and that would be more easily mantainable